enhancements/etcd/dr-leveraging-ceo-scaling.md

```diff
@@ -0,0 +1,188 @@
+
+TODO
+
+## Summary
+
+cluster-etcd-operator (CEO) is an operator that handles the scaling of etcd during cluster
+bootstrap and regular operation and provisioning etcd dependencies such as TLS certificates.
+
+## Motivation
+
+TODO
+
+
+### Goals
+
+1. Leverage the scale up mechanism of CEO
+
+### Non-Goals
+
+TODO
+
+## Challenges and possible alternatives
+
+A restore operation is a two step process on each node. 
+1. Restoring the data directory with appropriate environment variables with the following operation:
+```
+        ETCDCTL_API=3 etcdctl snapshot restore ./path/to/backup.db \
+                    --name $ETCD_NAME \
+                    --initial-cluster="$ETCD_NAME=$ETCD_NODE_PEER_URL" \
+                    --initial-cluster-token openshift-etcd \
+                    --initial-advertise-peer-urls $ETCD_NODE_PEER_UR \
+                    --data-dir="/path/to/new/data-dir"
+```
+2. Starting the etcd pod should be started with the same value of $ETCD_NAME, ETCD_INITIAL_CLUSTER and 
+$ETCD_NODE_PEER_URL as in step 1.
+
+This will allow etcd to populate the data directory with a new cluster ID, and new membership data that includes only
+1 member.
+
+Restoring a single member from a backup and scaling up has two distinct problem, they are:
+1. Get the first etcd member running. 
+2. Get the operator to deploy static pods that can decide that they are being restored, so they don't start on the 
+existing data directories.
+
+### Static Pod identifying a restore
+
+Assuming the first member is up, the control plane will be up and operator start running. A user can trigger a new 
+rollout of the etcd pods by running `oc patch forceredpeloymment=reason-time`. New pods will be rolled out on the 
+remaining masters that have to be restored. Before etcd commands on these pods are run, the existing data directory 
+needs to be removed, because they contain the previous cluster state information.
+
+During the start up pod queries etcd membership, if the pod is an unstarted member, it can be assumed that the data 
+directory present is from previous cluster and moved out.
+
+Other solution is to check the cluster id of the new cluster with the cluster id of the existing data directory, move 
+out the data directory only if they are different. This is a little time consuming to implement and test.
+
+### Getting the first etcd member running
+
+This will involve bringing a control plane up with a single node etcd cluster and then scaling it up to three. Before
+the restore operation is carried out, it is mandatory for the users to have a backup. If they do not have a backup, the 
+will lead fatal, non recoverable loss. The steps are:
+  1. move the existing etcd pod yaml out of kubelet manifest directory and move etcd data directory into a different 
+  location on all the master nodes. This will stop all the etcds leading to total quorum loss and remove all of the data
+   directories, making the backup a single source of truth.
```

Comment:
@alaypatel07 Is the first step necessary on rest of the master nodes? I was hoping the user could run most of these operations from one master host.

Reply:
chat on slack why it is necessary https://coreos.slack.com/archives/C851TKLLQ/p1582656368029000